### PR TITLE
Fixes ruins repeating

### PR DESCRIPTION
**Pull request by AnturK**
Yes.
@KorPhaeron

### CODE REVIEW

File: code/modules/awaymissions/zlevel.dm [hunk 1]
```diff
@@ -105,18 +105,20 @@ var/global/list/potentialRandomZlevels = generateMapList(filename = "config/away
...
+			possible_ruins += T
+	world << "<span class='boldannounce'>Loading ruins...</span>"
+	if(!template && possibleRuins.len)
+		template = possibleRuins[pick(possibleRuins)]
+	if(!template)
 		world << "<span class='boldannounce'>No ruins found.</span>"
 		return
-
+	template.load(get_turf(src),centered = TRUE)
+	template.loaded += 1
```
Comment (duncathan) [inline code/modules/awaymissions/zlevel.dm]: ++

File: code/modules/awaymissions/zlevel.dm [hunk 2]
```diff
@@ -105,18 +105,20 @@ var/global/list/potentialRandomZlevels = generateMapList(filename = "config/away
...
-
-	else
+	var/list/possible_ruins = list()
+	for(var/A in potentialRuins)
+		var/datum/template/T = A
+		if(!T.loaded)
+			possible_ruins += T
+	world << "<span class='boldannounce'>Loading ruins...</span>"
+	if(!template && possibleRuins.len)
+		template = possibleRuins[pick(possibleRuins)]
```
Comment (KazeEspada) [inline code/modules/awaymissions/zlevel.dm]: Just use pick and take here? and you wont need the var?

File: code/modules/awaymissions/zlevel.dm [hunk 3]
```diff
@@ -105,18 +105,20 @@ var/global/list/potentialRandomZlevels = generateMapList(filename = "config/away
...
-
-	else
+	var/list/possible_ruins = list()
+	for(var/A in potentialRuins)
+		var/datum/map_template/T = A
+		if(!T.loaded)
+			possible_ruins += T
+	world << "<span class='boldannounce'>Loading ruins...</span>"
+	if(!template && possible_ruins.len)
+		template = possible_ruins[pick(possible_ruins)]
```
Comment (KazeEspada) [inline code/modules/awaymissions/zlevel.dm]: Just use pick and take?

Comment (AnturK) [inline code/modules/awaymissions/zlevel.dm]: I'd still need to track it outside of this object. And this keeps tracking of usage as responsibility of the template.

**Comment by KorPhaeron** [comment]
Either I did something wrong/got really unlucky with ruin placement, or this prevents them  loading at all. I will try again in the morning.

**Pull request merged**